src/dialer/index.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const DialerQueue = require('./queue')
+const getPeerInfo = require('../get-peer-info')
+
+module.exports = function (_switch) {
+  const dialerQueue = new DialerQueue(_switch)
+
+  _switch.state.on('STOPPING:enter', abort)
+
+  /**
+   * @param {object} options
+   * @param {PeerInfo} options.peerInfo
+   * @param {string} options.protocol
+   * @param {boolean} options.useFSM If `callback` should return a ConnectionFSM
+   * @param {function(Error, Connection)} options.callback
```

Comment:
isn't it weird to have the callback in the `options` object?

Reply:
haha, yes, you're correct. I think it makes sense for it to go into the actual queue like that. Options really isn't the right term, it's really a DialRequest.